main.go

```diff
@@ -32,8 +33,25 @@ import (
 	"github.com/spf13/cast"
 
 	"github.com/bank-vaults/secret-init/provider"
+	"github.com/bank-vaults/secret-init/provider/file"
 )
 
+func NewProvider(providerName string) (provider.Provider, error) {
```

Comment:
Suggestion: Move `provider/provider.go` to `models/provider.go`.
This way, you can define `NewProvider` in a (new) `provider/provider.go` and avoid cyclic imports.